src/engine/source/indexerconnector/src/indexerQueryBuilder.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Wazuh - Indexer query builder.
+ * Copyright (C) 2015, Wazuh Inc.
+ * Nov 4, 2024.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _INDEXER_QUERY_BUILDER_HPP
+#define _INDEXER_QUERY_BUILDER_HPP
+
+#include <string>
+
+#include <nlohmann/json.hpp>
+
+#include <base/utils/builder.hpp>
+
+/**
+ * @brief IndexerQueryBuilder class.
+ *
+ * This class is a builder for creating OpenSearch queries.
+ *
+ * @note This class is not thread-safe. Implements the Builder pattern.
+ *
+ */
+class IndexerQueryBuilder : public base::utils::patterns::Builder<IndexerQueryBuilder>
```

Comment:
This class does not comply with the builder pattern, the final string should be built in the call to the build method.

The rest of the methods should be popular with members of the class.

Reply:
The builder method was removed.
We have been doing this since a long time ago.